### PR TITLE
feat: dynamically add workspaces in action

### PR DESCRIPTION
**Pull request by ArzelaAscoIi**
### Related Issues

- no

### Proposed Changes?
- run tests on new workspace

### How did you test it?
- running the CI 

### Notes for the reviewer
- We still have the 403 bug after creating a workspace.  Once this is fixed this will work too
- https://github.com/deepset-ai/deepset-cloud-sdk/actions/runs/6781847627/job/18432902291?pr=129

### Checklist

- [ ] I have updated the referenced issue with new insights and changes
- [ ] If this is a code change, I have added unit tests
- [ ] I've used the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title
- [ ] I updated the docstrings
- [ ] If this is a code change, I added meaningful logs and prepared Datadog visualizations and alerts

### CODE REVIEW

File: .github/workflows/deploy-test.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ env:
 
 jobs:
   deploy-test:
-    if: ${{ github.event.label.name == 'test-deploy' }}`
+    if: ${{ github.event.label.name == 'test-deploy' }} || github.event.label.name !='integration'`
```
Comment (wochinge) [inline .github/workflows/deploy-test.yml:14]: why do we need the `||` condition?

File: tests/integration/service/test_integration_files_service.py [hunk 2]
```diff
@@ -11,11 +11,6 @@
 from deepset_cloud_sdk._service.files_service import DeepsetCloudFile, FilesService
 
 
-@pytest.fixture
```
Comment (wochinge) [inline tests/integration/service/test_integration_files_service.py:14]: why can we drop this? Cause we're using the same workspace for both?

File: .github/workflows/deploy-test.yml (hunk 1 above)
Comment (ArzelaAscoIi) [reply .github/workflows/deploy-test.yml:14]: we don't want to build the test pypi if we added the integration label

File: tests/integration/service/test_integration_files_service.py (hunk 2 above)
Comment (ArzelaAscoIi) [reply tests/integration/service/test_integration_files_service.py:14]: yes

**Review by wochinge — APPROVED** [review]
Thanks for adding this!

**Pull request merged**